lib/itk/queue/connection.ex

```diff
@@ -61,31 +90,38 @@ defmodule ITKQueue.Connection do
     [host: host, port: port, username: username, password: password, heartbeat: heartbeat]
   end
 
+  # This will block the GenServer until connection is opened successfully.
   defp connect(params) do
-    AMQP.Connection.open(params)
+    params
+    |> AMQP.Connection.open()
     |> handle_connection_result(params)
   end
 
-  defp handle_connection_result({:ok, connection}, _params) do
-    Process.link(connection.pid)
-    connection
+  defp handle_connection_result({:ok, connection}, params) do
+    Process.monitor(connection.pid)
```

Comment:
Is it intentional to monitor the connection instead of linking it to the current process?

Reply:
Yes monitor is typically used for this scenario. We don't control the erlang amqp connection itself so we should not link to it.